rocket_sqlite_oracle/src/main.rs

```diff
@@ -0,0 +1,251 @@
+        limit 1;
+        "#,
+        timestamp,
+        timestamp,
+        timestamp
+    )
+    .fetch_one(&mut *conn)
+    .await
+    .unwrap();
+    let res = USDTickerResponse {
+        timestamp: row.timestamp.try_into().unwrap(),
+        ask_price: row.ask_price.try_into().unwrap(),
+        bid_price: row.bid_price.try_into().unwrap(),
+        signature: secp256k1::Signature::from_compact(&row.signed_message).unwrap(),
+    };
+
+    Json(res)
+}
+
+/// DB stuffs
+async fn run_migrations(rocket: Rocket<Build>) -> fairing::Result {
+    match ExampleDB::fetch(&rocket) {
+        Some(db) => match sqlx::migrate!("./migrations").run(&**db).await {
+            Ok(_) => Ok(rocket),
+            Err(e) => Err(rocket),
+        },
+        None => Err(rocket),
+    }
+}
+
+async fn kraken_ticker(rocket: Rocket<Build>) -> fairing::Result {
+    let db = match ExampleDB::fetch(&rocket) {
+        Some(db) => (**db).clone(),
+        None => return Err(rocket),
+    };
+
+    let key = State::<SecretKey>::get(&rocket).unwrap();
+    let key = (**key).clone();
+
+    tokio::spawn(async move {
+        let price_ticker_ws_url = Url::parse("wss://ws.kraken.com").unwrap();
+        let mut ticker = kraken::connect(price_ticker_ws_url)
+            .context("Failed to connect to kraken")
+            .unwrap();
+
+        loop {
+            match ticker.wait_for_next_update().await.unwrap() {
+                Ok(update) => {
+                    let signature = Message::new(update.ask.millicents, update.timestamp)
+                        .sign(&key)
+                        .serialize_compact()
+                        .to_vec();
+                    let timestamp = i64::try_from(update.timestamp).unwrap();
+                    let ask = i64::try_from(update.ask.millicents).unwrap();
+                    let bid = i64::try_from(update.bid.millicents).unwrap();
+                    sqlx::query!(
+                        r"
+                        insert into signatures (
+                          timestamp,
+                          ask_price,
+                          bid_price,
+                          signed_message
+                        ) values (?, ?, ?, ?)",
+                        timestamp,
+                        ask,
+                        bid,
+                        signature,
+                    )
+                    .execute(&db)
+                    .await
+                    .unwrap();
+                }
+                Err(e) => println!("Error: {:#}", e),
+            }
+        }
+    });
+
+    Ok(rocket)
+}
+
+/// CLI stuffs
+#[derive(Debug, StructOpt)]
+struct Cli {
+    /// Path to the file that contains the secret key of the oracle secret key
+    #[structopt(long)]
+    oracle_secret: PathBuf,
+
+    /// Set this flag to generate a secret file at the path specified by the
+    /// --oracle-secret argument
+    #[structopt(long)]
+    generate_oracle_secret: bool,
+}
```

Comment:
With these names, I would again question how useful it is to repeat the name "oracle" when the binary is already named "rocket_sqlite_oracle".